### PR TITLE
Avoid conversions Symbol -> String -> Symbol

### PR DESCRIPTION
**Pull request by fingolfin**
These lead to unnecessary allocations. Not much, but it adds up.

### CODE REVIEW

**Comment by fingolfin** [comment]
@HereAround @HechtiDerLachs this touches some of your code, so may wish to have a look (and possibly approve / request changes)

**Comment by thofma** [comment]
Maybe we now get a codecov report

**Review by HechtiDerLachs — APPROVED** [review]
Looks reasonable. Is the `VarName` type documented somewhere?

File: src/Rings/mpoly-localizations.jl [hunk 1]
```diff
@@ -2346,10 +2346,10 @@ quintuple ``(A, I, d, ϕ, θ)`` consisting of
 function as_affine_algebra(
     L::MPolyLocRing{BRT, BRET, RT, RET, 
 			     MPolyPowersOfElement{BRT, BRET, RT, RET}}; 
-    inverse_name::String="θ"
+    inverse_name::VarName=:_0
   ) where {BRT, BRET, RT, RET}
   R = base_ring(L)
-  A, phi, t = _add_variables_first(R, [inverse_name])
+  A, phi, t = _add_variables_first(R, [Symbol(inverse_name)])
```
Comment (thofma) [inline src/Rings/mpoly-localizations.jl:2352]: Does that function accept `Vector{Symbol}` instead of `Vector{String}`?

Comment (thofma) [inline src/Rings/mpoly-localizations.jl:2352]: Does that function accept `Vector{Symbol}` instead of `Vector{String}`?

**Review by HereAround — APPROVED** [review]
Thank you @fingolfin . Looks good to me!

File: src/Rings/mpoly-localizations.jl (hunk 1 above)
Comment (fingolfin) [reply src/Rings/mpoly-localizations.jl:2352]: IIRC this PR changes it to do so

**Review by thofma — APPROVED** [review]

File: src/Rings/mpolyquo-localizations.jl [hunk 2]
```diff
@@ -1324,14 +1324,14 @@ end
...
-function _add_variables(R::RingType, v::Vector{String}) where {RingType<:MPolyRing}
+function _add_variables(R::RingType, v::Vector{<:VarName}) where {RingType<:MPolyRing}
   ext_R, _ = polynomial_ring(coefficient_ring(R), vcat(symbols(R), Symbol.(v)))
   n = ngens(R)
   phi = hom(R, ext_R, gens(ext_R)[1:n])
   return ext_R, phi, gens(ext_R)[(n+1):ngens(ext_R)]
 end
 
-function _add_variables_first(R::RingType, v::Vector{String}) where {RingType<:MPolyRing}
+function _add_variables_first(R::RingType, v::Vector{<:VarName}) where {RingType<:MPolyRing}
```
Comment (fingolfin) [inline src/Rings/mpolyquo-localizations.jl:1334]: @thofma eg here

File: src/Rings/mpoly-localizations.jl (hunk 1 above)
Comment (thofma) [reply src/Rings/mpoly-localizations.jl:2352]: Ah, I found it. Thanks

**Pull request merged**